week-4/Homework/mandatory/2-writers.js

```diff
@@ -53,19 +53,41 @@ Exercise 1:
   insert corresponding values to the place holder that are indicated in courle braces:
   "Hi, my name is {firstName} {lastName}. I am {age} years old, and work as a {occupation}."
 */
-
-
+for (let key in writers)
+  console.log(
+    `Hi, my name is ${writers[key].firstName} ${writers[key].lastName}. I am ${writers[key].age} years old, and work as a ${writers[key].occupation}.`
+  );
 
 /*
 Exercise 2:
   Only `console.log()` the writers who are in their 40s with us anymore with the sentence:(meaning between 40 and 49)
   "Writer {firstName} {lastName} died at {age} years old."
 */
-
-
+for (let i = 0; i < writers.length; i++) {
+  if (
+    writers[i].age >= 40 &&
+    writers[i].age < 49 &&
+    writers[i].alive === false
+  ) {
+    console.log(
+      `Writer ${writers[i].firstName} ${writers[i].lastName} died at ${writers[i].age} years old.`
+    );
+  }
+}
 
 /*
 Exercise 3:
   Only `console.log()` contemporary writers who are in their forties:
   "Hi, my name is {firstName} {lastName}. I am {age} years old."
 */
+for (let i = 0; i < writers.length; i++) {
+  if (
+    writers[i].age >= 40 &&
+    writers[i].age < 49 &&
+    writers[i].alive === true
+  ) {
+    console.log(
+      `Hi,my name is ${writers[i].firstName} ${writers[i].lastName}. I am ${writers[i].age} years old.`
+    );
+  }
+}

```

Comment:
This is correct but an alternative is to use the filter method:
writers.filter(writer => !writer.alive && writer.age >= 40 && writer.age <= 49).forEach(writer => {
  console.log(`Writer: ${writer.firstName} ${writer.lastName} died at ${writer.age} years old.`);
});